libsol/system_instruction.c

```diff
@@ -16,26 +17,48 @@ enum SystemInstructionKind {
     AssignWithSeed
 };
 
-int parse_system_instruction_kind(Parser* parser, enum SystemInstructionKind* kind) {
+typedef struct SystemTransferInfo {
+    Pubkey* from;
+    Pubkey* to;
+    uint64_t lamports;
+} SystemTransferInfo;
+
+static int parse_system_instruction_kind(Parser* parser, enum SystemInstructionKind* kind) {
     return parse_u32(parser, (uint32_t *) kind);
 }
 
-int parse_system_transfer(Instruction* instruction, Pubkey* pubkeys, size_t pubkeys_length, Pubkey** from, Pubkey** to, uint64_t* lamports) {
+static int parse_system_transfer_instruction(Instruction* instruction, Pubkey* pubkeys, size_t pubkeys_length, SystemTransferInfo* info) {
     Parser parser = {instruction->data, instruction->data_length};
 
     enum SystemInstructionKind kind;
     BAIL_IF(parse_system_instruction_kind(&parser, &kind));
     BAIL_IF(kind != Transfer);
-    BAIL_IF(parse_u64(&parser, lamports));
+    BAIL_IF(parse_u64(&parser, &info->lamports));
 
     BAIL_IF(instruction->accounts_length < 2);
     uint8_t from_index = instruction->accounts[0];
     BAIL_IF(from_index >= pubkeys_length);
-    *from = &pubkeys[from_index];
+    info->from = &pubkeys[from_index];
 
     uint8_t to_index = instruction->accounts[1];
     BAIL_IF(to_index >= pubkeys_length);
-    *to = &pubkeys[to_index];
+    info->to = &pubkeys[to_index];
+
+    return 0;
+}
+
+// Returns 0, the fee payer, and system transfer info if provided a transfer message, otherwise non-zero.
```

Comment:
How does this return the fee payer? Just tacitly as header->pubkeys[0]?
Perhaps update this comment to be more clear.

Reply:
I failed to update the comment after changing the code. I'll fix that!

Reply:
Fixed in f3c1309902022404aae86c28f56df5659f7985b8